corehq/apps/users/tests/util.py

```diff
@@ -20,8 +21,21 @@ def create_usercase(user):
     )
 
 
-patch_user_data_db_layer = patch('corehq.apps.users.user_data.UserData.lazy_init',
-                                 new=lambda u, d: UserData({}, None, d))
+class patch_user_data_db_layer(ContextDecorator):
+    def __init__(self, user_schema=None):
+        self.user_schema = user_schema or {}
```

Comment:
nit: it looks like this `user_schema` param is unreferenced.

Reply:
As in no code currently references it? `self.user_schema` does get referenced in the `__enter__` block, but yes -- no test code is currently providing a custom user schema. However, I felt this was meant to be a general purpose test utility, and it felt likely that a test would want the ability to provide its own schema rather than just always assume it was empty. This didn't feel like a case of YAGNI to me, because I felt if this was not implemented and someone wanted to do this, it was fairly likely they'd reinvent the wheel or decide that the effort wasn't worth it and just skip the test.

Reply:
Yeah, I meant that no test code uses the custom schema.  I was thinking that if I needed that in my test, I'd probably use a DB test anyways rather than a mock, but I have no objections to keeping it in if you think it's useful.